code/game/sound.dm

```diff
@@ -21,12 +20,21 @@
 	for(var/P in listeners)
 		var/mob/M = P
 		if(get_dist(M, turf_source) <= maxdistance)
-			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, channel, pressure_affected, S)
+			sound_or_datum(M, turf_source, input, vol, vary, frequency, falloff, channel, pressure_affected)
 	for(var/P in SSmobs.dead_players_by_zlevel[z])
 		var/mob/M = P
 		if(get_dist(M, turf_source) <= maxdistance)
-			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, channel, pressure_affected, S)
+			sound_or_datum(M, turf_source, input, vol, vary, frequency, falloff, channel, pressure_affected)
+
+/proc/sound_or_datum(mob/receiver, turf/turf_source, input, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+	if(istype(input,/datum))
+		var/datum/outputs/O = input
+		O.send_info(receiver, turf_source, vol, vary, frequency, falloff, channel, pressure_affected)
+	else
+		var/sound/S = sound(get_sfx(input))
```

Comment:
I know I'm late to this party, but this is disgusting.

The whole reason `var/sound/S = sound(get_sfx(soundin))` was where it was, was so that you could create a sound *once* and cache it locally, then modify it for each client  for falloff. 

Creating sounds is quite expensive in BYOND, and it quickly becomes obvious the more mobs you have. While the z-level improvements have, no doubt, helped with this, it still doesn't solve the underlying problem. 

You just probably haven't noticed it yet since the sound range was dramatically nerfed quite a while ago.

Try removing the client limitation on `playsound` and put in like 100 mobs in the same tile and play a sound near them. Take now how expensive it is.